### DISCUSSION TITLE
userpref.config for dunst/waybar

### DISCUSSION THREAD

**Discussion by se7en-x230** [prasanthrangan/hyprdots#863] [Ideas]
each git pull and ./install.sh -r updates my hyperdots config. 
As I understand userpref.config keeps the userspecific settings of hyperland.
Would the same be possible for dunst and waybar?

I do some small color tweaks in dunst and like the 24h waybar format + mpris module.

Each time I update I have to do the same adjustments. I can script that to do the adjustments each time I run the update but I was wondering if there was the ability to do so with sourcing a similiar userpref file

**Comment by kRHYME7** [comment]
use ``` Hyprdots restore config ``` instead to manage  what files to be copied from the main repo. But first run ``` Hyprdots update ``` to pull changes.

Here is how I modified mine to just Sync the files I needed from the repo and Preserves my on changes
```
P|${HOME}|.zshrc .p10k.zsh|zsh oh-my-zsh-git zsh-theme-powerlevel10k pokemon-colorscripts-git
P|${HOME}/.config/Code/User|settings.json|visual-studio-code-bin
S|${HOME}/.config|dunst|dunst
S|${HOME}/.config/hypr|animations.conf hyprland.conf windowrules.conf scripts themes wallbash|hyprland
P|${HOME}/.config/hypr|theme.ctl|hyprland
P|${HOME}/.config/hypr|keybindings.conf monitors.conf userprefs.conf|hyprland
P|${HOME}/.config/hypr|nvidia.conf|hyprland nvidia-utils
S|${HOME}/.config|kitty|kitty
S|${HOME}/.config|Kvantum|kvantum
S|${HOME}/.config|MangoHud|mangohud
S|${HOME}/.config|neofetch|neofetch
S|${HOME}/.config|qt5ct|qt5ct
S|${HOME}/.config|rofi|rofi
S|${HOME}/.config|swaylock|swaylock-effects
P|${HOME}/.config|swww|swww
S|${HOME}/.config|config.jsonc style.css modules|waybar
P|${HOME}/.config|config.ctl|waybar
P|${HOME}/.config|wlogout|wlogout
P|${HOME}/.config|nwg-look gtk-3.0 xsettingsd|nwg-look
P|${HOME}/.icons|default|nwg-look
P|${HOME}|.gtkrc-2.0|nwg-look
S|${HOME}/.config|dolphinrc kdeglobals|dolphin
S|${HOME}/.local/share|dolphin|dolphin
S|${HOME}/.local/share/kxmlgui5|dolphin|dolphin
S|${HOME}/.config|lsd|lsd
S|${HOME}/.config|xsettingsd nwg-look gtk-3.0|nwg-look
```
O   Overwrite     Move Dots to Backup > Populate dots from Source Directory
S   Sync          Copy Dots to Backup > Copy to overwrite Dots from Source Directory      
P   Preserve      Populates Dots But Preserves User existing Dots
B   Backup        Only Backup the selected Dots
I   Ignore        Skip/Ignore the Dots


You can also use ```  Hyprdots restore defaults  ``` to mark files. 

About your request. 
It would definitely need a rewrite and be cluttered with  Userprefs for each configuration. (Trust me  I attempted once).

**Comment by se7en-x230** [comment]
So making 
I|${HOME}/.config|dunst|dunst 
would Ignore dunstrc
meaning marking waybar with an "I" would not get any changes OR would only not overwrite files that got edited?

Example: I edit clock.jsonc - this is kept the way I edited it.
                 you add or change network.jsonc - this gets updated

**Reply by kRHYME7** [reply]
I recommend using S and P for this. P if you want the Clone repo not touching a file/directory

```
O Overwrite Move Dots to Backup > Populate dots from Source Directory
S Sync Copy Dots to Backup > Copy to overwrite Dots from Source Directory
P Preserve Populates Dots But Preserves User existing Dots
B Backup Only Backup the selected Dots
I Ignore Skip/Ignore the Dots
```

     The Source is the repository of hyprdots, which contains the dotfiles you may want to apply to your system. The Target is your current configuration directory, where your existing dotfiles reside. The Backup Directory is where your existing dotfiles are stored when you choose to back them up, which is typically located at $HOME/.config/cfg_backups.

* Overwrite (O): This moves the existing dotfiles from the target directory to the backup directory and then populates the target directory with the new dotfiles from the source directory. This ensures that the target directory matches exactly what's in the source directory, potentially overwriting any customizations you had in place.

* Sync (S): This copies the dotfiles from the source directory to the target directory, without moving the originals. If a file already exists in the target, it will be overwritten; otherwise, it will be copied.

* Preserve (P): This also copies the dotfiles from the source to the target, but if a file already exists in the target, it will be preserved, meaning the existing file will not be touched. New files from the source that do not exist in the target will be populated.

* Backup Only (B): This only backs up the dotfiles from the target directory to the backup directory. No changes are made to the target or source directories.

* Ignore (I): This option does nothing with the dotfiles—they are left untouched in both the source and target directories.

**Reply by kRHYME7** [reply]
Be reminded that everytime you do this at least you have a backup for the current dots 


> Example: I edit clock.jsonc - this is kept the way I edited it.
you add or change network.jsonc - this gets updated

As you can see some are files and some are directories, therefore to preserve your clock configuration make a copy of it then PRESERVE the waybars config.ctl 

Lets focus on waybar part 
```
S|${HOME}/.config|config.jsonc style.css modules|waybar
P|${HOME}/.config|config.ctl|waybar
```
I use S for this file config.jsonc style.css modules ``` now clock.jsonc is under modules right ?( and it needs a longer list )
Therfore giving it S flag will overwrite the configs, but the files that do not have the same name will still be there so renaming your clock.jsonc into something like ``` myclock.jsonc ``` will make preserve your configuration 

Then use P PRESERVE on ``` config.ctl  ``` this one is the waybar module placements so it is good to Preserve this . 

example: 

```
1|30|bottom|( myclock  battery custom/gpuinfo )( hyprland/workspaces hyprland/window )|( wlr/taskbar mpris )|( tray )( network bluetooth pulseaudio pulseaudio#microphone )( idle_inhibitor custom/updates )
```

**Comment by se7en-x230** [comment]
Okey that makes a lot more sense now. And yes the easiest would be to just create a copy of with myclock and call that module.

Thanks for your help again
